### ISSUE TITLE
torch batchwise max with indices

### ISSUE THREAD

**Issue by sbagchi12** [pytorch/pytorch#32998]
## 🚀 Feature
A batchwise verison of torch.max() or torch.argmax().

## Motivation
I have an array with elements coming from multiple examples where each example contributes an independent number of elements. I would like to have an efficient way to find the max element along with the index.

## Pitch

torch.max() should support batches of max operations instead of a single tensor

## Alternatives

A hacky alternative 

c = torch.stack([torch.argmax(a[batch==i],dim=0,keepdim=True) for i in range(2)],dim=0)

**Comment by zou3519** [comment]
@sbagchi12 Could you give an example (with numbers) for what the inputs are and what you want as output? I'm having trouble following the description.

**Comment by epraun** [comment]
@zou3519 torch.group_max(values=torch.Tensor([1, 2, 3]), group_indices=torch.Tensor([0, 0, 1])) --> [2, 3]

**Comment by davidireland3** [comment]
@zou3519 I'm not sure if OP ever replied but I came across this post and I'm looking for a feature that will take a batch of matrices, i.e. a tensor of shape (B, M, M) and return the B maximums of each M by M matrix.

**Comment by ngoctnq** [comment]
@davidireland3 couldn't you just do `torch.max(mat.view(mat.size(0), -1), dim=1)`?
